mail_tracking_mailgun/models/res_partner.py

```diff
@@ -35,16 +36,24 @@ def _email_bounced_set(self, reason, event):
     @api.multi
     def check_email_validity(self):
         """Checks mailbox validity with Mailgun's API"""
-        api_key, api_url, domain = self.env[
+        api_key, api_url, domain, validation_key = self.env[
             'mail.tracking.email']._mailgun_values()
+        if not validation_key:
+            raise UserError(_('You need to configure mailgun.validation_key'
+                              ' in order to be able to check mails validity'))
         for partner in self:
             res = requests.get(
                 "%s/address/validate" % api_url,
                 auth=("api", api_key), params={
                     "address": partner.email,
                     "mailbox_verification": True,
                 })
-            if not res['mailbox_verification']:
+            if not res or res.status_code != 200:
+                raise UserError(_(
+                    'Error %s trying to '
+                    'check mail' % res.status_code or 'of connection'))
+            content = json.loads(res.content, res.apparent_encoding)
+            if content['mailbox_verification'] != 'true':
```

Comment:
Are you sure it returns true as a string? It's a JSON after all...
I think a link to api docs would be nice in the README

Reply:
Well it's kinda weird indeed, but that's what's returned: https://documentation.mailgun.com/en/latest/api-email-validation.html#example

I'm seeing I should test 'unkown' state also, anyway.

Reply:
And this kind of weird things is good to have links as comments for avoiding wondering if it's correct or not.

Reply:
Noted. I've documented it and improved the method cases.